code/modules/telesci/telesci_computer.dm

```diff
@@ -41,74 +43,72 @@
 	y_off = rand(-10,10)
 	x_player_off = 0
 	y_player_off = 0
-	if(ticker)
+	if (ticker && ticker.mode == GAME_STATE_PLAYING)
 		initialize()
 
 /obj/machinery/computer/telescience/initialize()
 	..()
-	if(!ticker)
-		cell=new/obj/item/weapon/cell(src)
+	if (!ticker || ticker.current_state != GAME_STATE_PLAYING)
+		cell = new/obj/item/weapon/cell(src)
 		cell.charge = 0
-	telepad = locate() in range(src, 7)
+
+	var/obj/machinery/telepad/possible_telepad = locate() in range(src, 7)
+	if (!possible_telepad.linked)
+		telepad = possible_telepad
+		telepad.linked = src
+
+/obj/machinery/computer/telescience/Destroy()
+	if (telepad)
+		telepad.linked = null
+		telepad = null
+
+	..()
 
 /obj/machinery/computer/telescience/process()
 	if(!cell || (stat & (BROKEN|NOPOWER)) || !anchored)
 		return
-	if(cell.give(100))
-		use_power(200)		//this used to use CELLRATE, but CELLRATE is fucking awful. feel free to fix this properly!
-	src.updateUsrDialog()
+
+	var/used = cell.give(100)
+	if (used)
+		use_power(used * 2) // This used to use CELLRATE, but CELLRATE is fucking awful. feel free to fix this properly!
+		nanomanager.update_uis(src)
 
 /obj/machinery/computer/telescience/attackby(obj/item/weapon/W, mob/user)
 	if(..())
-		return 1
+		return TRUE
 
-	if(stat & BROKEN)
-		return
+	if(stat & BROKEN || !ispowercell(W) || !anchored)
+		return FALSE
 
-	if(istype(W, /obj/item/weapon/cell) && anchored)
-		if(cell)
-			to_chat(user, "<span class='warning'>There is already a cell in \the [name].</span>")
-			return
-		else
-			if(areaMaster.power_equip == 0) // There's no APC in this area, don't try to cheat power!
-				to_chat(user, "<span class='warning'>\The [name] blinks red as you try to insert the cell!</span>")
-				return
+	if(cell)
+		to_chat(user, "<span class='warning'>There is already a cell in \the [name].</span>")
+		return TRUE
+
+	if(user.drop_item(W, src))
+		cell = W
+		user.visible_message("[user] inserts a cell into \the [src].", "You insert a cell into \the [src].")
```

Comment:
Shoudln't it be `\The [user]` ?

Reply:
No, `\The` is implied so it doesn't matter, we only do it for readability, and in most cases user is a proper noun.